### PR TITLE
Allow users to choose their own colour #2

### PR DESCRIPTION
**Pull request by tcmcgee**
added simple colourpicker! Tried to follow your style. Cool project!

sample

### CODE REVIEW

File: RedditCrap/app/html/popup.html [hunk 1]
```diff
@@ -30,8 +30,32 @@
...
+            <input type="radio" class="radio-color" name="color" id="radioColor" value="blue" />
+            <img src="../res/blue.png">
+        </label>
+        <label>
+          <input type="radio" class="radio-color" name="color" id="radioColor" value="green" />
+          <img src="../res/green.png">
+      </label>
+      <label>
+          <input type="radio" class="radio-color" name="color" id="radioColor" value="black" />
+          <img src="../res/black.png">
```
Comment (The-Quill) [inline RedditCrap/app/html/popup.html]: Use SVG instead of adding unnecessary image resources.

**Comment by tcmcgee** [comment]
@The-Quill How's that?

**Comment by The-Quill** [comment]
awesome

**Comment by Vannevelj** [comment]
Thank you for the work -- that looks great! There is another PR (https://github.com/Vannevelj/RedditCrap/pull/8) currently that updates the general UI layout and should make it a lot nicer (and you won't have to stick to my own ugly style). I will merge that one first since it makes the most changes -- perhaps you can adapt your PR to take that into account?

I believe there is a `<input type="color">` picker -- perhaps that might be better to use considering it allows you to change between any possible colour.

**Comment by tcmcgee** [comment]
@Vannevelj Hey I tried to get the colourpicker going, but there seemed to be some issues with it opening and the popup losing focus, so I decided to go back to the old way (with radio buttons). I changed it to work with the new CSS, and if you want to open a new issue (changing the colourpicker radio buttons to an input type color) I would be happy to work on that later. Thanks!

**Comment by tcmcgee** [comment]
No problem

**Comment by tcmcgee** [comment]
Sure I'll change the JSON later tonight. I'm not the most familiar with JS conventions either so feel free to critique me where you see fit!

**Comment by tcmcgee** [comment]
Went ahead and made changes you requested, and merged with master. Let me know if there's anything else. Perhaps a new issue for RGB colourpicker that I can get started on?

**Comment by Vannevelj** [comment]
Seems to work great! One addition though: if you open the popup and select a colour it will show which colour is selected. If you now close it and open it again, the colour is unselected. Can you make sure that when the popup is opened, the appropriate box is selected? 

I'll create a separate issue for a colour picker.

**Comment by tcmcgee** [comment]
Good catch! How's that?

File: RedditCrap/app/js/popup.js [hunk 2]
```diff
@@ -52,6 +52,19 @@ function displaySavedSettings() {
 
         var selectElement = document.getElementById('crappyAction');
         selectElement.selectedIndex = data.crappyAction;
+
+    var radios = document.forms["radio-form"].elements["colour"];
+    var button;
+    for(var i = 0, max = radios.length; i < max; i++)
```
Comment (Vannevelj) [inline RedditCrap/app/js/popup.js]: Seems a bit verbose -- `for(var i = 0; i < radios.length; i++)` is the way most people expect it to be written I think.

File: RedditCrap/app/js/popup.js [hunk 3]
```diff
@@ -52,6 +52,19 @@ function displaySavedSettings() {
...
         var selectElement = document.getElementById('crappyAction');
         selectElement.selectedIndex = data.crappyAction;
+
+    var radios = document.forms["radio-form"].elements["colour"];
+    var button;
+    for(var i = 0, max = radios.length; i < max; i++) 
+    {
+        if (radios[i].value == data.crappyColour)
+        {
+            radios[i].checked = true;
```
Comment (Vannevelj) [inline RedditCrap/app/js/popup.js]: We can add a `break;` behind this considering there should be only 1 colour selected at a time and save some precious clock cycles :-)

**Comment by tcmcgee** [comment]
There you go, thanks for pointing that out.

**Comment by Vannevelj** [comment]
Perfect! I appreciate your patience. Last night I released version 1.1 which includes an improved UI, a small bugfix and your colour picker (but without this last little bug) -- I will include those changes in the next release.

**Pull request merged**